assets/js/editor-components/color-panel/index.tsx

```diff
@@ -0,0 +1,182 @@
+		const colorObject = getColorObject( colors, colorValue, context ) || {};
+		setAttributes( {
+			[ colorName ]: colorObject,
+		} );
+	};
+};
+
+const createSettings = (
+	colorTypes: CustomColorsMap,
+	colors: ColorPaletteOption[] & GradientPaletteOption[],
+	attributes: ColorAttributes | undefined,
+	setAttributes: ( attributes: Record< string, unknown > ) => void
+) => {
+	return Object.entries( colorTypes ).reduce(
+		( settingsAccumulator, [ colorAttributeName, colorAttribute ] ) => {
+			const colorSetter = createSetColor(
+				colorAttributeName,
+				colorAttribute.context,
+				colors,
+				setAttributes
+			);
+			const colorSetting = {
+				colorValue:
+					attributes?.[ colorAttributeName ]?.color ?? undefined,
+				label: colorAttribute.label,
+				onColorChange: colorSetter,
+				resetAllFilter: () => colorSetter(),
+			};
+			settingsAccumulator.push( colorSetting );
+			return settingsAccumulator;
+		},
+		[] as ColorSetting[]
+	);
+};
+
+export const ColorPanel = ( {
+	colorTypes,
+}: {
+	colorTypes: CustomColorsMap;
+} ) => {
+	const colorGradientOptions = useMultipleOriginColorsAndGradients();
+	const flattenedColors = flattenColors( colorGradientOptions );
+	const { clientId } = useBlockEditContext();
+	const attributes = useSelect(
+		( select ) => {
+			// @ts-ignore @wordpress/block-editor/store types not provided
+			const { getBlockAttributes } = select( blockEditorStore );
+			return getBlockAttributes( clientId ) || {};
+		},
+		[ clientId ]
+	);
+	// @ts-ignore @wordpress/block-editor/store types not provided
+	const { updateBlockAttributes } = useDispatch( blockEditorStore );
+	const settings = useMemo( () => {
+		return createSettings(
+			colorTypes,
+			flattenedColors,
+			attributes,
+			( newAttributes ) =>
+				updateBlockAttributes( clientId, newAttributes )
+		);
+	}, [
+		colorTypes,
+		flattenedColors,
+		updateBlockAttributes,
+		attributes,
+		clientId,
+	] );
+	return (
+		colorGradientOptions.hasColorsOrGradients && (
+			// @ts-ignore The dev package version doesn't have types for group.
+			<InspectorControls group="color">
+				<ColorGradientSettingsDropdown
+					__experimentalIsRenderedInSidebar
+					settings={ settings }
+					panelId={ clientId }
+					{ ...colorGradientOptions }
+				/>
+			</InspectorControls>
+		)
+	);
+};
+
+export const useColorPanelStyles = ( colorTypes: CustomColorsMap ) => {
+	const { clientId } = useBlockEditContext();
+	const attributes = useSelect(
+		( select ) => {
+			// @ts-ignore @wordpress/block-editor/store types not provided
+			const { getBlockAttributes } = select( blockEditorStore );
+			return getBlockAttributes( clientId ) || {};
+		},
+		[ clientId ]
+	);
+
+	const className: string[] = [];
+
+	Object.keys( colorTypes ).forEach( ( colorName ) => {
+		const attribute = attributes?.[ colorName ];
+		if ( attribute && attribute?.color ) {
+			className.push( `has-${ colorTypes[ colorName ].context }` );
+		}
+	} );
+
+	return {
+		className,
+	};
+};
```

Comment:
Quite subjective, but posting it just in case you agree. :slightly_smiling_face:  I would move this hook to a separate file, `use-color-panel-styles.ts` or something like this. I think that one export per-file makes code easier to read and to find via search.

Reply:
Will update this based on the outcome of this discussion below https://github.com/woocommerce/woocommerce-blocks/pull/10062#discussion_r1265325651